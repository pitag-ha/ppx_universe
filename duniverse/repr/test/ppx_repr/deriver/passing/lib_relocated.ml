module Elsewhere : sig
  module Foo : module type of Repr

  type t [@@deriving repr { lib = Some "Foo" }]
end = struct
  module Foo = Repr
  module Irmin = struct end

  type t = unit * unit [@@deriving repr { lib = Some "Foo" }]
end

module Locally_avaliable : sig
  type 'a ty
  type t [@@deriving repr { lib = None }]
end = struct
  let pair, unit = Repr.(pair, unit)

  type 'a ty = 'a Repr.ty

  module Irmin = struct end

  type t = unit * unit [@@deriving repr { lib = None }]
end
