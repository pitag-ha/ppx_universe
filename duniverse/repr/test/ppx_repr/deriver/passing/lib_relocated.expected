module Elsewhere :
  sig
    module Foo : module type of Repr
    type t[@@deriving repr { lib = (Some "Foo") }]
    include sig val t : t Foo.t end[@@ocaml.doc "@inline"][@@merlin.hide ]
  end =
  struct
    module Foo = Repr
    module Irmin = struct  end
    type t = (unit * unit)[@@deriving repr { lib = (Some "Foo") }]
    include struct let t = Foo.pair Foo.unit Foo.unit end[@@ocaml.doc
                                                           "@inline"]
    [@@merlin.hide ]
  end 
module Locally_avaliable :
  sig
    type 'a ty
    type t[@@deriving repr { lib = None }]
    include sig val t : t ty end[@@ocaml.doc "@inline"][@@merlin.hide ]
  end =
  struct
    let (pair, unit) = let open Repr in (pair, unit)
    type 'a ty = 'a Repr.ty
    module Irmin = struct  end
    type t = (unit * unit)[@@deriving repr { lib = None }]
    include struct let t = pair unit unit end[@@ocaml.doc "@inline"][@@merlin.hide
                                                                    ]
  end 
